submissions/YarHadi/task_js_oop/app.js

```diff
@@ -0,0 +1,146 @@
+const fieldWidth = 505,
+  fieldHeight = 606,
+  columnsNumber = 5,
+  rowNumber = 7,
+  cellWidth = fieldWidth / columnsNumber,
+  cellHeight = fieldHeight / rowNumber,
+  startPlayerX = fieldWidth / 2 - cellWidth / 2,
+  startPlayerY = fieldHeight / 2 + cellHeight,
+  leftBorder = 0,
+  rightBorder = fieldWidth - cellWidth,
+  topBorder = 0,
+  playerHitbox=40;
+
+// Enemies our player must avoid
+
+var Enemy = function (x, y) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+
+  this.x = x;
+  this.y = y;
+  this.speed = Math.random() * 200 + 200;
+
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+
+  this.sprite = "images/enemy-bug.png";
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+
+Enemy.prototype.update = function (dt) {
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+
+  this.x += this.speed * dt;
+  if (this.x > ctx.canvas.width) {
+    this.x = leftBorder;
+  }
+
+  this.collision(player);
```

Comment:
`player` here is still a global variable. 

The hint in the requirements says:

> Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.

How would you interpret it?
Where passing a game object as an argument to other game objects happens?